bundles/org.palladiosimulator.commons/src/org/palladiosimulator/commons/eclipseutils/ExtensionHelper.java

```diff
@@ -176,32 +176,69 @@ private ExtensionHelper() {
      * @param filterAttributeValue
      *            the atrribute's value to be used for filtering. Found attributes have to equal
      *            this value in case they should be chosen.
-     * @return an executable extension matching to all selection criteria defined by this method's
-     *         parameters.
+     * @return a list of executable extensions matching to all selection criteria defined by this 
+     *         method's parameters.
      * @param <DATA_TYPE>
      *            the data type of the executable extension.
      */
-    public static <DATA_TYPE> DATA_TYPE getExecutableExtension(final String extensionPointID, final String elementName,
-            final String attributeName, final String filterAttributeName, final String filterAttributeValue) {
+    public static <DATA_TYPE> List<DATA_TYPE> getExecutableExtensions(final String extensionPointID,
+            final String elementName, final String attributeName, final String filterAttributeName,
+            final String filterAttributeValue) {
         final List<IExtension> extensions = loadExtensions(extensionPointID);
+        final List<DATA_TYPE> results = new LinkedList<DATA_TYPE>();
 
         for (final IExtension extension : extensions) {
             final IConfigurationElement configurationElement = obtainConfigurationElement(extension, elementName);
 
-            if (configurationElement.getAttribute(filterAttributeName).equals(filterAttributeValue)) {
+            if (configurationElement.getAttribute(filterAttributeName)
+                .equals(filterAttributeValue)) {
                 try {
                     @SuppressWarnings("unchecked")
                     final DATA_TYPE executableExtension = (DATA_TYPE) configurationElement

```

Comment:
I cannot find a check before this cast. Wouldn't it be better to test for the type and throw an exception if the type does not match?

Reply:
The cast is done based on the generic type parameter. There is no way of type checking that. We would need to adapt the signature to include an additional class object parameter. I would rather omit that.

Reply:
Added additional signatures which allow safe-casting and deprecated the current single extension case.